src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/judgmentonline/SendJudgmentDetailsCjesHandler.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.civil.handler.callback.camunda.judgmentonline;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.camunda.bpm.engine.RuntimeService;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.Callback;
+import uk.gov.hmcts.reform.civil.callback.CallbackHandler;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.callback.CaseEvent;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.service.judgments.ReportJudgmentsService;
+import java.util.List;
+import java.util.Map;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.SEND_JUDGMENT_DETAILS_CJES;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.SEND_JUDGMENT_DETAILS_CJES_SA;
+import static uk.gov.hmcts.reform.civil.enums.YesOrNo.YES;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SendJudgmentDetailsCjesHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = List.of(SEND_JUDGMENT_DETAILS_CJES,
+                                                          SEND_JUDGMENT_DETAILS_CJES_SA);
+    public static final String TASK_ID = "SendJudgmentDetailsToCJES";
+    private final ObjectMapper objectMapper;
+    private final RuntimeService runTimeService;
+    private final ReportJudgmentsService reportJudgmentsService;
+
+    @Override
+    protected Map<String, Callback> callbacks() {
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT), this::generatePayload
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    CallbackResponse generatePayload(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        CaseData.CaseDataBuilder<?, ?> caseDataBuilder = caseData.toBuilder();
+        CaseEvent caseEvent = CaseEvent.valueOf(callbackParams.getRequest().getEventId());
+
+        if (YES.equals(caseData.getJoIsRegisteredWithRTL())) {
```

Comment:
This varaible doesnt get set "JoIsRegisteredWithRTL" when its default judgment/JBA as this variable is only part of UI screen for Judgment recorded after hearing. We need to check this "isRegisterWithRTL" values from active or historic judgment as it will be always populated for any type of judgment.

Reply:
I've updated it, would be worth rechecking to see the new logic is correct @Omaira-Melo-Hmcts @kalachandrasekar1